File: src/echo-actors/src/main/scala/echo/actor/directory/repository/PodcastRepository.scala
package echo.actor.directory.repository

import echo.core.domain.entity.{PodcastEntity}
import echo.core.domain.feed.FeedStatus
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.{JpaRepository, Query}
import org.springframework.data.repository.query.Param

/**
  * @author Maximilian Irro
  */
trait PodcastRepository extends JpaRepository[PodcastEntity, java.lang.Long] {

    def findOneByEchoId(echoId: String): PodcastEntity

    @Query("SELECT DISTINCT podcast FROM PodcastEntity podcast " +
           "LEFT JOIN podcast.feeds feed " +
           "WHERE feed.echoId = :feedExo")
    def findOneByFeed(@Param("feedExo") feedExo: String): PodcastEntity

    def findByRegistrationCompleteTrue(pageable: Pageable): java.util.List[PodcastEntity]

    @Query("SELECT count(podcast) FROM PodcastEntity podcast")
    def countAll(): Long

    @Query("SELECT count(podcast) FROM PodcastEntity podcast WHERE podcast.registrationComplete = true")
    def countAllRegistrationCompleteTrue(): Long
}

File: src/echo-microservices/app-catalog/src/main/java/echo/microservice/catalog/repository/PodcastRepository.java
package echo.microservice.catalog.repository;

import echo.core.domain.entity.PodcastEntity;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

/**
 * @author Maximilian Irro
 */
public interface PodcastRepository extends JpaRepository<PodcastEntity,Long> {

    PodcastEntity findOneByExo(String exo);

    @Query("SELECT DISTINCT podcast FROM PodcastEntity podcast " +
            "LEFT JOIN podcast.feeds feed " +
            "WHERE feed.exo = :feedExo")
    PodcastEntity findOneByFeed(@Param("feedExo") String feedExo);

    List<PodcastEntity> findByRegistrationCompleteTrue(Pageable pageable);

    @Query("SELECT count(podcast) FROM PodcastEntity podcast")
    Long countAll();

    @Query("SELECT count(podcast) FROM PodcastEntity podcast WHERE podcast.registrationComplete = true")
    Long countAllRegistrationCompleteTrue();

}
